src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -118,12 +135,14 @@ public static IResolveFieldContext<TSourceType> As<TSourceType>(this IResolveFie
             if (context == null)
                 throw new ArgumentNullException(nameof(context));
 
-            if (context.OutputExtensions == null || context.OutputExtensions.Count == 0)
-                return null;
+            return GetByPath(context.OutputExtensions, path, true);
+        }
 
-            lock (context.OutputExtensions)
+        private static object? GetByPath(IDictionary<string, object?>? dictionary, string path, bool useLock)
```

Comment:
I would suggest changing the signature here to `IReadOnlyDictionary` here and the `is IDictionary` line.  It probably makes little difference, as the `ImmutableDictionary` class and even the brand-new `FrozenDictionary` class supports `IDictionary` as well as `IReadOnlyDictionary`.

Reply:
done